container/go/pkg/oci/write_test.go

```diff
@@ -0,0 +1,157 @@
+//////////////////////////////////////////////////////////////////////
+// Tests for Write function.
+
+package oci
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/google/go-containerregistry/pkg/v1/validate"
+	"github.com/pkg/errors"
+)
+
+var (
+	blob     = "blobs/"
+	testPath = "testdata/test_write1"
+)
+
+var testCases = []struct {
+	// name is the name of the test case.
+	name string
+	// dataPath is where the test data is found.
+	dataPath string
+	// path to the image manifest relative to the dataPath.
+	manifestPath string
+	// content is the OCI layout content where the key is the file path and value is the sha256 sum.
+	// it contains a mapping of the sha256 code of the different files inside the OCI layout.
+	expectedContent map[string]v1.Hash
+}{
+	{
+		"alpine_linux",
+		"testdata/test_write1/small_linux.tar",
+		"sha256/9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c",
+		map[string]v1.Hash{
+			"blobs/sha256/9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c": v1.Hash{Algorithm: "sha256", Hex: "9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c"},
+			"index.json": v1.Hash{Algorithm: "sha256", Hex: "ce6ed2e6817add8a5a0e94f4c2ac00360c7ca2cc98c4972b86530f14de622508"},
+			"oci-layout": v1.Hash{Algorithm: "sha256", Hex: "b66dbb27a73334db6ac9c030475837bd7f4472d835c72b2360534b203edce6cb"},
+		},
+	},
+}
+
+// TestWrite checks to ensure that the v1.Image written is in OCI format by validating image output format and content.
+func TestWrite(t *testing.T) {
+	for _, wt := range testCases {
+		t.Run(wt.name, func(t *testing.T) {
+			tmp, err := ioutil.TempDir("", "write-oci-index-test")
+			if err != nil {
+				t.Errorf("Unable to create a TempDir %s: %v", os.TempDir(), err)
+			}
+
+			defer os.RemoveAll(tmp)
+
+			img, err := tarball.ImageFromPath(wt.dataPath, nil)
+			if err != nil {
+				t.Errorf("Error loading image from testdata: %v", err)
+			}
+
+			if err := Write(img, tmp); err != nil {
+				t.Errorf("Write(%s): %v", tmp, err)
+			}
+
+			if err := assertValidImageOCILayout(tmp, img, wt.expectedContent, wt.manifestPath); err != nil {
+				t.Errorf("Image index written is not a valid OCI format: %v", err)
+			}
+		})
+	}
+}
+
+// assertValidImageOCILayout checks that the index written is not corrupt
+// and that the contents of the OCI layout are all present and non-corrupt.
+func assertValidImageOCILayout(ociDir string, img v1.Image, content map[string]v1.Hash, shaPth string) error {
+	written, err := layout.ImageIndexFromPath(ociDir)
+	if err != nil {
+		return errors.Wrapf(err, "failed to write image to temp path %s", ociDir)
+	}
+
+	if err := validate.Index(written); err != nil {
+		return errors.Wrapf(err, "validate.Index() =")
```

Comment:
error messages in Go should be standalone statements on their own and have as much context as possible. Here I would suggest saying:
`errors.Wrapf(err, "failed to validate the image written to %s using validate.Index()", ociDir)`